impl.go

```diff
@@ -305,6 +320,32 @@ func validReceiver(recv string) bool {
 	return err == nil
 }
 
+// commentsBefore returns true if comments precede the field. It returns false
+// otherwise.
+func commentsBefore(field *ast.Field, commentGroups []*ast.CommentGroup) bool {
+	if len(commentGroups) > 0 {
+		return commentGroups[0].Pos() < field.Pos()
+	}
+	return false
+}
+
+// flattenCommentMap flattens the comment map to a string.
+func flattenCommentMap(m ast.CommentMap) string {
+	var result strings.Builder
+	for _, cmmtgroups := range m {
+		for _, cmmtgroup := range cmmtgroups {
+			for _, cmt := range cmmtgroup.List {
+				result.WriteString(cmt.Text)
+				// add an end-of-line character if this is '//'-style comment
+				if cmt.Text[0] == '/' {
```

Comment:
Should this be `cmt.Text[1]`?

Reply:
I am trying to determine here if the comments we are dealing with is //-style comments. If they are, `Text` property will have `\n` character removed (please see [this](https://golang.org/src/go/ast/ast.go?#L64)). So I am add `\n` character for consistent formatting.

Reply:
Right. I must be missing something. ISTM that there are two kinds of comments. One starts `//` and the other starts `/*`. Each has `/` as `Text[0]`; `Text[1]` would be the way to tell them apart.

Reply:
Oops! You are right absolutely. I meant `1` believe me or not! 😄